test/regression/dollar-sign-body-test.coffee

```diff
@@ -0,0 +1,26 @@
+{assert} = require 'chai'
+Gavel2Html = require('../../src/index')
+
+describe 'Dollar sign in the body', ->
+  gavelResult = {
+    fields: {
+      body: {
+        kind: 'json',
+        values: {
+          actual: JSON.stringify({ symbol: '$' }),
+          expected: JSON.stringify({ symbol: 'a' }),
+        },
+        errors: []
+      }
+    }
+  }
+
+  payload = {
+    fieldName: 'body',
+    fieldResult: gavelResult.fields.body,
+  }
+  diff = new Gavel2Html payload
+  html = diff.getHtml()
+
+  it 'escapes the dollar sign in the diff', ->
+    assert.equal(html, '<li>{</li>\n<li>  &quot;symbol&quot;: &quot;$$&quot;</li>\n<li>}</li>')
```

Comment:
I see double dollar '$$' sign when I put this to the browser location:
```
data:text/html,<li>{</li><li>  &quot;symbol&quot;: &quot;$$&quot;</li><li>}</li>
```

Therefore I wonder whether it shouldn't really be:
```suggestion
    assert.equal(html, '<li>{</li>\n<li>  &quot;symbol&quot;: &quot;$&quot;</li>\n<li>}</li>')
```

Reply:
I wonder if this is related to coffee-script. A single `$` evaluates to some `#{this}` in the expected string, failing the test. It basically becomes:

```js
assert.equal(html, '<li>...#{this}</li>')
```

During the test run. Do you know what `#{this}` even refers to?

Reply:
Hmm, https://coffeescript-cookbook.github.io/chapters/strings/interpolation

Reply:
Strange with your '$' experience. Single dollar seems to work for me in https://jsfiddle.net/0nb6uh7L/

Reply:
@artem-zakharchenko The following patch works for me. What I am missing?

```
$ git diff
diff --git a/src/json-result-converter.coffee b/src/json-result-converter.coffee
index 6941e2c..e7c9a83 100644
--- a/src/json-result-converter.coffee
+++ b/src/json-result-converter.coffee
@@ -156,7 +156,7 @@ class JsonResultConverter extends Converter
             .replace(/"/g, '\"')
             # Replace the "$" character in the value to prevent it
             # from forming a string replacement sequence ($&, $1, etc.).
-            .replace('$', '$$$')
+            .replace('$', '$$')
           )
         s += postStringValue if postStringValue
         s += '&quot;' # sanitizeData('"')
diff --git a/test/regression/dollar-sign-body-test.coffee b/test/regression/dollar-sign-body-test.coffee
index 6576bb8..5cca84e 100644
--- a/test/regression/dollar-sign-body-test.coffee
+++ b/test/regression/dollar-sign-body-test.coffee
@@ -23,4 +23,4 @@ describe 'Dollar sign in the body', ->
   html = diff.getHtml()
 
   it 'escapes the dollar sign in the diff', ->
-    assert.equal(html, '<li>{</li>\n<li>  &quot;symbol&quot;: &quot;$$&quot;</li>\n<li>}</li>')
+    assert.equal(html, '<li>{</li>\n<li>  &quot;symbol&quot;: &quot;$&quot;</li>\n<li>}</li>')
```

Reply:
You're right, those changes do work. I think I had a wrong escape when I tried to change the expected string in a test, which made the test fail. Thank you for pointing out 👍